python/ray/tests/test_k8s.py

```diff
@@ -0,0 +1,372 @@
+import os
+import subprocess
+import time
+import unittest
+
+import pytest
+
+from ray.autoscaler._private.kubernetes import core_api
+
+
+class KubernetesTest(unittest.TestCase):
+    def test_up_and_down(self):
+        """(1) Runs 'ray up' with a Kubernetes config that specifies
+        min_workers=1.
+        (2) Runs 'ray exec' to read monitor logs and confirm that worker and
+        head are connected.
+        (3) Runs 'ray down' and confirms that the cluster is gone."""
+        with open("k8s_test.yaml", "w") as f:
+            f.write(test_yaml)
+
+        # ray up
+        os.system(
+            "minikube start; ray up k8s_test.yaml --yes --no-config-cache")
+
+        # Check for two pods (worker and head).
+        while True:
+            pod_list = core_api().list_namespaced_pod(namespace="ray")
+            if len(pod_list.items) == 2:
+                print("Head and worker pods started.")
+                break
+            else:
+                print("Waiting for worker pod to start.")
+                time.sleep(1)
+
+        # Read logs with ray exec and check that worker and head are connected.
+        # (Since the config yaml is legacy-style, we check for
+        # ray-legacy-*-node_type.)
+        while True:
+            monitor_output = subprocess.check_output([
+                "ray", "exec", "k8s_test.yaml",
+                "tail -n 100 /tmp/ray/session_latest/logs/monitor*"
+            ]).decode()
+            output_ok = ("ray-legacy-head-node-type" in monitor_output
+                         and "ray-legacy-worker-node-type" in monitor_output)
+            if output_ok:
+                print("Worker node connected. Monitoring ok.")
+                break
+            else:
+                print("Waiting for worker to connect.")
+                time.sleep(1)
+
+        # ray down.
+        os.system("ray down k8s_test.yaml --yes")
+
+        # Check that there are no pods left in namespace ray to confirm that
+        # the cluster is gone.
+        while True:
+            pod_list = core_api().list_namespaced_pod(namespace="ray")
+            if len(pod_list.items) == 0:
+                print("Cluster shut down succesfully.")
+                break
+            else:
+                print("Waiting for cluster to shut down.")
+
+        os.remove("k8s_test.yaml")
+
+test_yaml = """
```

Comment:
can you move this to a separate directory in ray/tests, similar to typing_files and test_cli_patterns.